### PR TITLE
Add metadata link to gemspec

### PR DESCRIPTION
**Pull request by greysteil**
Specifying metadata links in the gemspec replaces specifying them through the Rubygems web UI (which will soon be deprecated).

### CODE REVIEW

**Comment by olleolleolle** [comment]
Can we find the links for _all_ of the supported keys?

```ruby
{
    "bug_tracker_uri" => "https://github.com/celluloid/celluloid/issues",
    "changelog_uri" => "https://github.com/celluloid/celluloid/blob/master/CHANGES.md",
    "documentation_uri" => "https://www.rubydoc.info/gems/celluloid",
    "homepage_uri" => "https://celluloid.io/",
    "mailing_list_uri" => "http://groups.google.com/group/celluloid-ruby",
    "source_code_uri" => "https://github.com/celluloid/celluloid",
    "wiki_uri" => "https://github.com/celluloid/celluloid/wiki"
  }
```

(Some, more careful folks would check if `s.respond_to?(:metadata)`. I wouldn't.)

**Comment by greysteil** [comment]
Now with all of the links.

File: sinatra-contrib/sinatra-contrib.gemspec [hunk 1]
```diff
@@ -18,6 +18,11 @@ Gem::Specification.new do |s|
     "ideas.md",
     "sinatra-contrib.gemspec"
   ]
+  s.metadata = {
+    'source_code_uri'   => 'https://github.com/sinatra/sinatra/tree/master/sinatra-contrib'
```
Comment (namusyaka) [inline sinatra-contrib/sinatra-contrib.gemspec]: Missing `,`

File: rack-protection/rack-protection.gemspec [hunk 2]
```diff
@@ -17,6 +17,11 @@ Gem::Specification.new do |s|
     "Gemfile",
     "rack-protection.gemspec"
   ]
+  s.metadata = {
+    'source_code_uri' => 'https://github.com/sinatra/sinatra/tree/master/rack-protection'
+    'homepage_uri'      => 'http://sinatrarb.com/protection/',
+    'documentation_uri' => 'https://www.rubydoc.info/gems/rack-protection'
+  }
```
Comment (namusyaka) [inline rack-protection/rack-protection.gemspec]: I'd like to raise an error with kindness message `unless s.respond_to?(:metadata)`. Should be something like:

```ruby
if s.respond_to?(:metadata)
  s.metadata = {}
else
  raise 'RubyGems 2.0 or newer is required to protect against public gem pushes.'
end
```

**Comment by greysteil** [comment]
@namusyaka - updated!

File: rack-protection/rack-protection.gemspec [hunk 3]
```diff
@@ -18,6 +18,21 @@ Gem::Specification.new do |s|
...
+      'homepage_uri'      => 'http://sinatrarb.com/protection/',
+      'documentation_uri' => 'https://www.rubydoc.info/gems/rack-protection'
+    }
+  else
+    msg = "RubyGems 2.0 or newer is required to protect against public "\
+          "gem pushes. You can update your rubygems version by running:\n\n"\
+          "gem install rubygems-update\n"\
+          "update_rubygems\n"\
+          "gem update --system"
+    raise msg
```
Comment (namusyaka) [inline rack-protection/rack-protection.gemspec]: No need to assign a local variable, I think.
Instead, you can use here-document.
Please take a look at the following code:

```ruby
  raise <<-EOF
RubyGems 2.0 or newer is required to protect against public
gem pushes. You can update your rubygems version by running:
  gem install rubygems-update
update_rubygems:
  gem update --system
  EOF
```

**Pull request merged**